app/src/main/java/org/luxoft/sdl_core/SdlLauncherService.java

```diff
@@ -135,41 +138,79 @@ private void showToastMessage(String message) {
-                .setContentTitle(SDL_CONTENT_TITLE)
-                .setContentText(text)
-                .setOnlyAlertOnce(true) // so when data is updated don't make sound and alert in android 8.0+
-                .setOngoing(true)
-                .setSmallIcon(R.drawable.ic_launcher_foreground)
-                .setContentIntent(contentIntent)
-                .build();
+        Intent intent = new Intent(this, MainActivity.class);
+        intent.addFlags(Intent.FLAG_ACTIVITY_CLEAR_TOP);
+        boolean isThreadAlive = sdl_thread_ != null && sdl_thread_.isAlive();
+        intent.putExtra(IS_SDL_SERVICE_THREAD_ALIVE, isThreadAlive);
+
+        PendingIntent contentIntent = PendingIntent.getActivity(this, 0, intent, PendingIntent.FLAG_UPDATE_CURRENT);
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
+            if (channel_id == null) {
+                Log.w(TAG, "Cannot create notification. Create and save channel_id firstly");
+                return null;
+            }
+            return new NotificationCompat.Builder(this, channel_id)
+                    .setContentTitle(SDL_CONTENT_TITLE)
+                    .setContentText(text)
+                    .setOnlyAlertOnce(true) // so when data is updated don't make sound and alert in android 8.0+
+                    .setOngoing(true)
+                    .setSmallIcon(R.drawable.ic_launcher_foreground)
+                    .setContentIntent(contentIntent)
+                    .build();
+        } else {
+            return new NotificationCompat.Builder(this)
+                    .setContentTitle(SDL_CONTENT_TITLE)
+                    .setContentText(text)
+                    .setOngoing(true)
+                    .setSmallIcon(R.drawable.ic_launcher_foreground)
+                    .setContentIntent(contentIntent)
+                    .build();
+        }
     }
 
-    @SuppressLint("NewApi")
-    public void enterForeground() {
+    private void enterForeground() {
         NotificationManager notificationManager = (NotificationManager) getSystemService(Context.NOTIFICATION_SERVICE);
-        if (notificationManager != null) {
+        if (notificationManager == null) {
+            Log.w(TAG, "NotificationManager are not available. Notification skipped.");
+            return;
+        }
+        final String text = "SDL core is starting...";
+
+        if (Build.VERSION.SDK_INT >= Build.VERSION_CODES.O) {
             if (channel_id == null) {
                 NotificationChannel channel = new NotificationChannel(APP_ID, SERVICE_NAME, NotificationManager.IMPORTANCE_DEFAULT);
                 notificationManager.createNotificationChannel(channel);
                 channel_id = channel.getId();
             }
 
-            startForeground(FOREGROUND_SERVICE_ID, getServiceNotification("SDL core is starting..."));
+            startForeground(FOREGROUND_SERVICE_ID, getServiceNotification(text));
+        } else {
+            updateServiceNotification(text);
```

Comment:
@YaroslavLutsenko if older android versions does not have a notification channel, this call probably does nothing. Maybe we can remove it?

Reply:
@AKalinich-Luxoft this method will show notification for older android versions as well. `updateServiceNotification(text);` with text "SDL core is starting...". 
I can check version in line 61 - 64 and call `enterForeground()` or `updateServiceNotification(text);` It will allow to delete `updateServiceNotification(text);` from the `enterForeground()`. Will it be better?

Reply:
@YaroslavLutsenko my question was just about `updateServiceNotification` in this `else` branch.
I am okay with the existing solution